### PR TITLE
test: add unit test for `pop build parachain`

### PR DESCRIPTION
**Pull request by brunopgalvao**
Unit test for `pop build parachain`

### CODE REVIEW

**Comment by brunopgalvao** [comment]
Not sure why the CI is failing. Works fine on my machine.

**Comment by weezy20** [comment]
@brunopgalvao setup protoc on the CI. Some dependency wants that.

**Comment by AlexD10S** [comment]
Build the parachain is very slow, and now running `cargo test` can take a while. I opened an Issue about that a while ago: https://github.com/r0gue-io/pop-cli/issues/18
Any thoughts on how to improve this?

File: src/engines/parachain_engine.rs [hunk 1]
```diff
@@ -80,35 +80,53 @@ pub fn build_parachain(path: &Option<PathBuf>) -> anyhow::Result<()> {
...
+		let temp_dir =
+			setup_template_and_instantiate().expect("Failed to setup template and instantiate");
 
 		// Verify that the generated chain_spec.rs file contains the expected content
 		let generated_file_content =
-			fs::read_to_string(temp_dir.path().join("node/src/chain_spec.rs"))?;
+			fs::read_to_string(temp_dir.path().join("node/src/chain_spec.rs"))
+				.expect("Failed to read file");
 		assert!(generated_file_content.contains("DOT"));
 		assert!(generated_file_content.contains("18"));
```
Comment (weezy20) [inline src/engines/parachain_engine.rs:108]: There can be any number of `18`s in the file. A part of a key for instance. Please improve this. 
`tokenDecimals: 18` is a much better check.

File: src/engines/parachain_engine.rs [hunk 2]
```diff
@@ -80,35 +80,53 @@ pub fn build_parachain(path: &Option<PathBuf>) -> anyhow::Result<()> {
...
+	fn test_parachain_instantiate_base_template() -> Result<(), Error> {
+		let temp_dir =
+			setup_template_and_instantiate().expect("Failed to setup template and instantiate");
 
 		// Verify that the generated chain_spec.rs file contains the expected content
 		let generated_file_content =
-			fs::read_to_string(temp_dir.path().join("node/src/chain_spec.rs"))?;
+			fs::read_to_string(temp_dir.path().join("node/src/chain_spec.rs"))
+				.expect("Failed to read file");
 		assert!(generated_file_content.contains("DOT"));
```
Comment (weezy20) [inline src/engines/parachain_engine.rs:107]: Same here. `tokenSymbol: "DOT"`

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -28,12 +28,21 @@ jobs:
...
       run: |
         git config --global user.name "${GITHUB_ACTOR}"
         git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
 
+    - name: Rust Setup
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: stable
+        target: wasm32-unknown-unknown
```
Comment (weezy20) [inline .github/workflows/build.yml:42]: Why add `wasm32` target to the CI?

**Review by weezy20 — DISMISSED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (brunopgalvao) [reply .github/workflows/build.yml]: Good point - we are not building Substrate here

File: src/engines/parachain_engine.rs (hunk 1 above)
Comment (brunopgalvao) [reply src/engines/parachain_engine.rs]: Done.

File: src/engines/parachain_engine.rs (hunk 2 above)
Comment (brunopgalvao) [reply src/engines/parachain_engine.rs]: Done.

File: src/engines/parachain_engine.rs [hunk 4]
```diff
@@ -80,35 +80,53 @@ pub fn build_parachain(path: &Option<PathBuf>) -> anyhow::Result<()> {
 #[cfg(test)]
 mod tests {
 	use super::*;
+	use anyhow::{Error, Result};
 	use std::{env::current_dir, fs};
 
-	#[test]
-	fn test_instantiate_template_dir_base() -> Result<(), Box<dyn std::error::Error>> {
-		let temp_dir = tempfile::tempdir()?;
+	fn setup_template_and_instantiate() -> Result<tempfile::TempDir, Error> {
```
[the remark this answers is not in the document]
Comment (brunopgalvao) [reply src/engines/parachain_engine.rs]: Done.

**Comment by brunopgalvao** [comment]
Looks like we do need the wasm32-unknown-unknown target:

```
   Compiling parachain-template-runtime v0.7.0 (/tmp/.tmp4LtFzj/runtime)
error: failed to run custom build command for `parachain-template-runtime v0.7.0 (/tmp/.tmp4LtFzj/runtime)`

Caused by:
  process didn't exit successfully: `/tmp/.tmp4LtFzj/target/release/build/parachain-template-runtime-bdf6ae77f659e646/build-script-build` (exit status: 1)
  --- stderr
  Cannot compile the WASM runtime: the `wasm32-unknown-unknown` target is not installed!
  You can install it with `rustup target add wasm32-unknown-unknown` if you're using `rustup`.
warning: build failed, waiting for other jobs to finish...
test engines::parachain_engine::tests::test_parachain_build_after_instantiating_template ... FAILED

failures:

---- engines::parachain_engine::tests::test_parachain_build_after_instantiating_template stdout ----
thread 'engines::parachain_engine::tests::test_parachain_build_after_instantiating_template' panicked at src/engines/parachain_engine.rs:129:9:
Result should be Ok
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    engines::parachain_engine::tests::test_parachain_build_after_instantiating_template

test result: FAILED. 5 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1358.77s

error: test failed, to rerun pass `--bin pop`
Error: Process completed with exit code 101.
```
Add it back in.

**Review by AlexD10S — APPROVED** [review]

**Pull request merged**